src/v/kafka/server/handlers/topics/types.cc

```diff
@@ -109,10 +109,20 @@ get_string_value(const config_map_t& config, std::string_view key) {
 static std::optional<bool>
 get_bool_value(const config_map_t& config, std::string_view key) {
     if (auto it = config.find(key); it != config.end()) {
-        return string_switch<std::optional<bool>>(it->second)
-          .match("true", true)
-          .match("false", false)
-          .default_match(std::nullopt);
```

Comment:
I think removing the `default_match` is fine. I know we specifically ignore unknown topic properties to aid migration from other kafka clusters. But I can't see a reason why we would want to ignore malformed properties for known topic properties.

Reply:
Yeah, these changes very specifically do not affect unknown topic properties- only properties we are actually attempting to parse.